github_helper.py

```diff
@@ -23,30 +22,26 @@ def __init__(self, org, repo):
         self.repo_url = GITHUB_REPO_FMT_URL.format(org, repo)
         with open(GITHUB_SECRET, 'r') as key_file:
             self.bot_key = key_file.read().strip()
-        _print_flush('Loaded key file.')
 
     def fetch_prs(self):
         """Gets the PRs for the configured repository.
 
         Returns:
-            An iterable containing instances of GithubPR if successful.
-            Empty list otherwise.
+            A tuple of (pr_list, error).
+            pr_list contains an iterable containing instances of GithubPR if
+            successful; empty list otherwise.
+            error is None if no errors, else error message.
         """
         try:
             prs = self.get(GITHUB_PULLS_ENDPOINT)
-            return [GithubPR(self, pr) for pr in prs]
+            return [GithubPR(self, pr) for pr in prs], None
         except HTTPError as exc:
-            _print_flush('Non-200 HTTP Code Received:')
-            _print_flush(exc)
+            return [], 'Non-200 HTTP Code Received: {}'.format(exc)
         except Timeout as exc:
-            _print_flush('Request timed out:')
-            _print_flush(exc)
+            return [], 'Request timed out: {}'.format(exc)
         except RequestException as exc:
-            _print_flush('Catastrophic error in requests:')
-            _print_flush(exc)
-        _print_flush('Retrieving pull requests failed.')
-        return []
-
+            return [], 'Catastrophic error in requests: {}'.format(exc)
+        return [], 'Unknown Error'
```

Comment:
I want to avoid polluting the caller with a million different exception handling cases though. In this case I think we're returning the right amount of information and catching the exceptions right where they happen instead of having them propagate up the stack. I see the value to having them do so, but I think that the information we're returning currently is approximately correct. The caller doesn't care if it was an HTTP error or an internal requests error, just wants a string saying what went wrong. When I add the new error type I'll probably swap over to raising that since that seems right-er to me, but for now I think this does what we want. I'll think a little more about this tonight but unless you have strenuous objections I'm inclined to leave as-is.